test/dynamo/test_ctx_manager.py

```diff
@@ -697,6 +697,33 @@ def fn(x, y, z):
 
         self.assertEqual(cnts.frame_count, 2)
 
+    def _graph_break_inlining_autocast_test_helper(self, device):
+        def gn(x, y):
+            with torch.autocast(device_type=device, dtype=torch.bfloat16):
+                z = torch.mm(x, y)
+                torch._dynamo.graph_break()
+                return torch.sin(z)
+
+        def fn(x, y):
+            z = torch.mm(x, y)
+            z = z + gn(x, y)
+            return z
+
+        x = torch.rand(3, 3).to(device)
+        y = torch.rand(3, 3).to(device)
+        opt_fn = torch.compile(backend="eager")(fn)
+        ref = fn(x, y)
+        res = opt_fn(x, y)
+        self.assertEqual(ref, res)
+
+    def test_graph_break_inlining_autocast(self):
+        for device in ["cuda", "cpu"]:
+            if device == "cuda" and not (
+                torch.cuda.is_available() and torch.cuda.is_bf16_supported()
+            ):
+                continue
+            self._graph_break_inlining_autocast_test_helper(device)
```

Comment:
Curious: with current impl, we will only run `_graph_break_inlining_autocast_test_helper` if CUDA is available and BF16 is supported. Do we always have CI machine that meets this requirement?

Reply:
Yes, we have many CI machines with CUDA and BF16 enabled. This check is used to skip CUDA/BF16 test cases when running on CPU only CI machines.